From 515bf0ac1926702509547d3ba14de78145116245 Mon Sep 17 00:00:00 2001
From: haiqiang yao <haiqiangx.yao@intel.com>
Date: Fri, 15 Jan 2016 16:53:08 +0800
Subject: [PATCH 05/13] atomisp: Remove default OB2 param for BTNS project

this patch will remove workaround
"I3094ea4c46d4a898427988afbbc93afa7825aa70".

Now the AIQ will take care of OB2 setting, so remove it.

Change-Id: I1b95f6450ac207236760a641468daac3d9cf188a
Original-Tracked_On: #H1504135600
Signed-off-by: haiqiang yao <haiqiangx.yao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/95792
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: Dessert M BXT RVP (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-17482
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://android.intel.com:443/464448
---
 .../media/pci/atomisp2/atomisp_driver/atomisp_cmd.c       | 15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index acddfa0..787afb6 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -3235,18 +3235,6 @@ void atomisp_apply_css_parameters(
 	if (css_param->update_flag.dvs_6axis_config)
 		atomisp_css_set_dvs_6axis(asd, css_param->dvs_6axis);
 
-	/* Workaround for BTNS, apply default OB2 setting for IMX227.
-	   before AIQ for ISP2.7 is available,  we have to keep such
-	   workaround for a reasonable image color
-	 */
-#ifdef CONFIG_EXTERNAL_BTNS_CAMERA
-	{
-		static struct atomisp_css_ob2_config imx227_ob2_config =
-			{1024, 1024, 1024, 1024};
-		atomisp_css_set_ob2_config(asd, &imx227_ob2_config);
-	}
-#endif
-
 	/* Add some parameters for isp2.7 */
 	if (css_param->update_flag.dpc2_config)
 		atomisp_css_set_dpc2_config(asd, &css_param->dpc2_config);
@@ -3254,6 +3242,9 @@ void atomisp_apply_css_parameters(
 	if (css_param->update_flag.eed1_8_config)
 		atomisp_css_set_eed1_8_config(asd, &css_param->eed1_8_config);
 
+	if (css_param->update_flag.ob2_config)
+		atomisp_css_set_ob2_config(asd, &css_param->ob2_config);
+
 	if (css_param->update_flag.ctc2_config)
 		atomisp_css_set_ctc2_config(asd, &css_param->ctc2_config);
 
-- 
1.9.1

